domain/infosync/info.go

```diff
@@ -338,76 +338,58 @@ func GetAllServerInfo(ctx context.Context) (map[string]*ServerInfo, error) {
 	return is.getAllServerInfo(ctx)
 }
 
-// UpdateTiFlashTableSyncProgress is used to update the tiflash table replica sync progress.
-func UpdateTiFlashTableSyncProgress(ctx context.Context, tid int64, progressString string) error {
-	is, err := getGlobalInfoSyncer()
-	if err != nil {
-		return err
-	}
-	if is.etcdCli == nil {
-		return nil
-	}
-	key := fmt.Sprintf("%s/%v", TiFlashTableSyncProgressPath, tid)
-	return util.PutKVToEtcd(ctx, is.etcdCli, keyOpDefaultRetryCnt, key, progressString)
-}
-
 // DeleteTiFlashTableSyncProgress is used to delete the tiflash table replica sync progress.
 func DeleteTiFlashTableSyncProgress(tableInfo *model.TableInfo) error {
 	is, err := getGlobalInfoSyncer()
 	if err != nil {
 		return err
 	}
-	if is.etcdCli == nil {
-		return nil
-	}
 	if pi := tableInfo.GetPartitionInfo(); pi != nil {
 		for _, p := range pi.Definitions {
-			key := fmt.Sprintf("%s/%v", TiFlashTableSyncProgressPath, p.ID)
-			err = util.DeleteKeyFromEtcd(key, is.etcdCli, keyOpDefaultRetryCnt, keyOpDefaultTimeout)
-			if err != nil {
-				return err
-			}
+			is.tiflashReplicaManager.DeleteTiFlashProgressFromCache(p.ID)
 		}
 	} else {
-		key := fmt.Sprintf("%s/%v", TiFlashTableSyncProgressPath, tableInfo.ID)
-		return util.DeleteKeyFromEtcd(key, is.etcdCli, keyOpDefaultRetryCnt, keyOpDefaultTimeout)
+		is.tiflashReplicaManager.DeleteTiFlashProgressFromCache(tableInfo.ID)
 	}
 	return nil
 }
 
-// GetTiFlashTableSyncProgress uses to get all the tiflash table replica sync progress.
-func GetTiFlashTableSyncProgress(ctx context.Context) (map[int64]float64, error) {
+// MustGetTiFlashProgress gets tiflash replica progress from tiflashProgressCache, if cache noe exist, get and insert progress into cache.
+func MustGetTiFlashProgress(tableID int64, replicaCount uint64, tiFlashStores *map[int64]helper.StoreStat) float64 {
```

Comment:
Is it possible to move this interface to TiFlash Manager? Seems that TiFlash manager does not need to expose the cache logic. The caching logic can be something internally.

In this way you may also reduce the public APIs of TiFlash manager (i.e. removing cache APIs)